shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -578,34 +578,35 @@ def _valid_dest_location_for_menu(self):
         domain = self._valid_dest_location_for_menu_domain()
         return self.env["stock.location"].search(domain)
 
-    def _set_quantity__check_location(self, move_line, location, confirmation=False):
+    def _set_quantity__check_location(self, move_line, location, confirmation=None):
         valid_locations_for_move_line = (
             self._set_quantity__valid_dest_location_for_move_line(move_line)
         )
         valid_locations_for_menu = self._valid_dest_location_for_menu()
         message = False
-        asking_confirmation = False
         if location in valid_locations_for_move_line:
             # scanned location is valid, return no response
             pass
         elif (
             location in valid_locations_for_menu
             and self.work.menu.allow_alternative_destination
         ):
-            # Considered valid if scan confirmed
-            if not confirmation:
+            if confirmation == location.barcode:
+                # Confirmation is valid, return no response
+                pass
+            else:
                 # Ask for confirmation
                 orig_location = move_line.location_dest_id
                 message = self.msg_store.confirm_location_changed(
                     orig_location, location
                 )
-                asking_confirmation = True
+                confirmation = location.barcode
```

Comment:
You have to compare with the scanned barcode (you need to add it in the method arguments)
You don't know how the location was matched. And this is also called when a package is scanned

Reply:
New edge case because of the improvement on the confirmation.
Indeed it can also be called when a package is scanned (the package is not empty so using the location).
But this will lead to an incoherent situation. The user scanned a package in a location that need confirmation so the user asked is being asked to confirm the location change by scanning the location again ? But the system will not remember it should be put in a package.

Reply:
And do we not have a bug on https://github.com/OCA/wms/pull/710/files#diff-f871450f795713d4ab1061b7117262fdee975730cb8f85f5963d06d339d7e39dL723
Because if the user is being asked to confirmed a location in the response. The `package_id` on the move.line should not yet be changed....

Reply:
The user scan a barcode and we ask to confirm by scanning the same barcode. You first put the scanned barcode as the value to confirm. You then need to compare the newly scanned barcode with the barcode that must be confirmed.
We don't care if the barcode is a package name, a location barcode, or whatever complex gs1 sscc label.
We just want the user to scan twice the same barcode.

Reply:
> And do we not have a bug on https://github.com/OCA/wms/pull/710/files#diff-f871450f795713d4ab1061b7117262fdee975730cb8f85f5963d06d339d7e39dL723 Because if the user is being asked to confirm a location in the response. The `package_id` on the move.line should not yet be changed....

No there is no such bug. The user scan a package that is on another location. We ask to confirm this location change. You set scanned barcode in `confirmation`. The user can confirm this change by scanning again the package.

Reply:
But the destination package is set on the move.line before the user has confirmed the action !

Reply:
yes, that is true.
I think we have to check that ask confirmation is not set in the response here (and there's no error) here.
This _set_quantity__check_location is part of the handlers.
https://github.com/OCA/wms/blob/0108725c0ca9b5332727e9144d17ead1006ff04e/shopfloor_single_product_transfer/services/single_product_transfer.py#L724
Regarding the barcode check, it can be done in the endpoint here https://github.com/OCA/wms/blob/0108725c0ca9b5332727e9144d17ead1006ff04e/shopfloor_single_product_transfer/services/single_product_transfer.py#L835
confirmation is true if barcode == confirmation, so you don't have to check this in the other methods.